### PR TITLE
Food refactor

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This was a huge pain in the butt, but it has to be done.

Refactors food/snacks and cleans up the code:

- Bitesize var is set on the object as opposed to in `New`
- Reagents now properly use `list_reagents` as opposed to `reagents.add_reagent()` in `New`
- Fixed a few improper reagent values on some candy (ie: some gummies had a total of 25 reagents when it was clear it was meant to be 12, since all the other subtypes were similar)
- Got rid of most relative pathing related to food I could find
- Fixed up some spans
- Reagent removal properly handled on the *recipe* as opposed to the food *object*
- Got rid of an old baychem pill that is unused (and the reagent it has doesn't even exist anymore)
- Removed a lot of implied `src`
- Removed a lot of implied `var`
- Removed a lot of `as X`

Basically this doesn't impact players at all--the single intentional deviation I made was....the *base* type of cotton candy have a bite size of 4 (as opposed to 3) so `bitesize = 4` didn't have to be copy-pasted for ***every*** single child of cotton candy (all the subtypes did, in fact, have a  bitesize value of 4).

### CODE REVIEW

**Comment by Crazylemon64** [comment]
> `I made the base type of cotton candy have a bite size of 4 (as opposed to 3)`

literally unplayable \</s>

**Comment by theColdflame** [comment]
Does this fix the maint donut?

File: code/modules/food_and_drinks/recipes/recipes_oven.dm [hunk 1]
```diff
@@ -293,6 +293,11 @@
 	)
 	result = /obj/item/weapon/reagent_containers/food/snacks/sliceable/bread
 
+/datum/recipe/oven/bread/make_food(obj/container)
+	var/obj/item/weapon/reagent_containers/food/snacks/sliceable/bread/being_cooked = ..(container)
```
Comment (Markolie) [inline code/modules/food_and_drinks/recipes/recipes_oven.dm]: No need to call container here, ..() automatically includes all of the procs parameters.

**Comment by fludd12** [comment]
With bite_size available in VV now, I can make people consume entire cakes or pies at once. Finally, adminbus + mint toxin will be a viable murder strategy.

Or, y'know, do something benevolent with it. Because I'm boring.

**Comment by Fox-McCloud** [comment]
@fludd12 

You are already able to do that, it's just that whoever initially made food set the var in `New` instead of on the object itself...which makes no sense.
